js/editor.js

```diff
@@ -1,425 +1,453 @@
-	is_editor_shown = true;
-	giveEditorFocus();
-}
+		_self = this;
 
-var is_editor_shown = false;
-$(document).ready(function () {
-	if ($('#isPublic').val()){
-		// disable editor in public mode (not supported yet)
-		return;
-	}
-	if (typeof FileActions !== 'undefined') {
-		FileActions.register('text', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
-		});
-		FileActions.setDefault('text', 'Edit');
-		FileActions.register('application/xml', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
-		});
-		FileActions.setDefault('application/xml', 'Edit');
-		FileActions.register('application/x-empty', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
-		});
-		FileActions.setDefault('application/x-empty', 'Edit');
-		FileActions.register('inode/x-empty', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
-		});
-		FileActions.setDefault('inode/x-empty', 'Edit');
-		FileActions.register('application/x-php', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
+		$.each(mimes, function(key, value) {
+			OCA.Files.fileActions.registerAction({
+				name: 'Edit',
+				mime: value,
+				actionHandler: _self._onEditorTrigger,
+				permissions: OC.PERMISSION_READ
+			});
+			OCA.Files.fileActions.setDefault(value, 'Edit');
 		});
-		FileActions.setDefault('application/x-php', 'Edit');
-		FileActions.register('application/javascript', 'Edit', OC.PERMISSION_READ, '', function (filename) {
-			showFileEditor($('#dir').val(), filename);
+
+	},
+
+	/**
+	 * Actually fire up the editor in a container
+	 */
+	loadEditor: function(container, file) {
+		var _self = this;
+		// Insert the editor into the container
+		container.html('<div id="editor_overlay"></div><div id="editor_container" class="icon-loading"><div id="editor"></div></div>');
+		$('#app-content').append(container);
+
+		// Get the file data
+		this.loadFile(
+			file.dir,
+			file.name,
+			function(file, data){
+				// Success!
+				// Sort the title
+				document.title = file.name + ' - ' + document.title;
+				// Load ace
+				$('#'+_self.editor).text(data);
+				// Configure ace
+				_self.configureACE(file);
+				// Show the controls
+				_self.loadControlBar(file, _self.currentContext);
+				//window.aceEditor.getSession().on('change', _self.setupAutosave);
+				window.aceEditor.focus();
+			},
+			function(message){
+				// Oh dear
+				OC.dialogs.alert(message, t('files_texteditor', 'An error occurred!'));
+			});
+	},
+
+	/**
+	 * Load the editor control bar
+	 */
+	loadControlBar: function(file, context) {
+		var html = '<small class="filename">'+file.name+'</small><button id="editor_save">' + t('files_texteditor', 'Save') + '</button><small class="lastsaved">'+t('files_texteditor', 'Last saved: never')+'</small>';
```

Comment:
:bomb: :bomb: :bomb: :bomb: :bomb: 

`file.name` is not escaped

:bomb: :bomb: :bomb: :bomb: :bomb: